kai/llm_interfacing/model_provider.py

```diff
@@ -13,6 +15,10 @@
 
 from kai.kai_config import KaiConfigModels
 
+if TYPE_CHECKING:
```

Comment:
Why this though?

Reply:
`TYPE_CHECKING` will only import the types during type checking to avoid circular imports. _Technically_ we should import types that we only use for type checking in one of these blocks.